libs/providers/openshift.py

```
@@ -192,8 +192,30 @@ def vm_dict(
+        devices: dict[str, Any] = domain.get("devices", {})
+        tpm_config: dict[str, Any] | None = devices.get("tpm")
+        firmware_info["tpm_present"] = bool(tpm_config.get("persistent")) if tpm_config else False
+
+        bootloader: dict[str, Any] | None = firmware_spec.get("bootloader") if firmware_spec else None
+        efi_config: dict[str, Any] | None = bootloader.get("efi") if bootloader else None
+
+        if efi_config:
+            firmware_info["boot_firmware"] = "efi"
+            secure_boot: bool | None = efi_config.get("secureBoot")
+            if secure_boot is None:
+                raise ValueError(f"secureBoot not found in EFI config for VM '{cnv_vm_name}'")
```

Ruff (0.15.17)

[warning] Avoid specifying long messages outside the exception class

(TRY003)